openwisp_notifications/tests/test_api.py

```diff
@@ -425,3 +432,45 @@ def test_obsolete_notifications_busy_worker(self, mocked_task):
             self.assertDictEqual(response.data, {'detail': NOT_FOUND_ERROR})
 
         unregister_notification_type('test_type')
+
+    def test_notification_redirect_api(self):
+        def _unread_notification(notification):
+            notification.unread = True
+            notification.save()
+
+        notify.send(sender=self.admin, type='default', target=self.admin)
+        notification = Notification.objects.first()
+
+        with self.subTest('Test non-existent notification'):
+            url = reverse(
+                f'{self.url_namespace}:notification_read_redirect', args=(uuid.uuid4(),)
+            )
+            response = self.client.get(url)
+            self.assertEqual(response.status_code, 404)
+            self.assertDictEqual(response.data, {'detail': NOT_FOUND_ERROR})
+
+        with self.subTest('Test existent notification'):
+            url = reverse(
+                f'{self.url_namespace}:notification_read_redirect',
+                args=(notification.pk,),
+            )
+            response = self.client.get(url)
+            self.assertEqual(response.status_code, 302)
+            self.assertEqual(response.url, notification.target_url)
+            notification.refresh_from_db()
+            self.assertEqual(notification.unread, False)
+
+        _unread_notification(notification)
+
+        with self.subTest('Test user not logged in'):
+            self.client.logout()
+            url = reverse(
+                f'{self.url_namespace}:notification_read_redirect',
+                args=(notification.pk,),
+            )
+            response = self.client.get(url)
+            self.assertEqual(response.status_code, 302)
+            self.assertEqual(
+                response.url,
+                f'/admin/login/?next=/api/v1/notifications/{notification.pk}/redirect/',
```

Comment:
would have been better to use `reverse()` for consistency.
It's also good to use `reverse()` for maintainability, although this form is more readable.
Not worth it to debate these minor details though :grin: